src-java/testing/functional-tests/src/test/groovy/org/openkilda/functionaltests/spec/server42/Server42RttSpec.groovy

```diff
@@ -279,6 +332,200 @@ class Server42RttSpec extends HealthCheckSpecification {
+        def flowRttFeatureStartState = changeFlowRttToggle(true)
+        def initialSwitchRtt = [].collectEntries { [fl1SwPair.src, changeFlowRttSwitch(fl1SwPair.src, true)] }
+
+        and: "Two flows on the given switch pairs"
+        def flowCreateTime = new Date()
+        //take shorter flowid due to https://github.com/telstra/open-kilda/issues/3720
+        def flow1 = flowHelperV2.randomFlow(fl1SwPair).tap { it.flowId = it.flowId.take(25) }
+        def flow2 = flowHelperV2.randomFlow(fl2SwPair).tap { it.flowId = it.flowId.take(25) }
+        flowHelperV2.addFlow(flow1)
+        flowHelperV2.addFlow(flow2)
+
+        //make sure stats for the flow1 in forward directions are available and not available for the flow2
+        Wrappers.wait(STATS_FROM_SERVER42_LOGGING_TIMEOUT, 1) {
+            assert !otsdb.query(flowCreateTime, metricPrefix + "flow.rtt",
+                    [flowid   : flow1.flowId,
+                     direction: "forward"]).dps.isEmpty()
+            assert otsdb.query(flowCreateTime, metricPrefix + "flow.rtt",
+                    [flowid   : flow2.flowId,
+                     direction: "forward"]).dps.isEmpty()
+        }
+
+        when: "Try to swap src endpoints for two flows"
+        def flow1Src = flow2.source
+        def flow1Dst = flow1.destination
+        def flow2Src = flow1.source
+        def flow2Dst = flow2.destination
+        def response = northbound.swapFlowEndpoint(
+                new SwapFlowPayload(flow1.flowId, flow1Src, flow1Dst),
+                new SwapFlowPayload(flow2.flowId, flow2Src, flow2Dst))
+
+        then: "Endpoints are successfully swapped"
+        with(response) {
+            it.firstFlow.source == flow1Src
+            it.firstFlow.destination == flow1Dst
+            it.secondFlow.source == flow2Src
+            it.secondFlow.destination == flow2Dst
+        }
+
+        def flow1Updated = northboundV2.getFlow(flow1.flowId)
+        def flow2Updated = northboundV2.getFlow(flow2.flowId)
+        flow1Updated.source == flow1Src
+        flow1Updated.destination == flow1Dst
+        flow2Updated.source == flow2Src
+        flow2Updated.destination == flow2Dst
+
+        and: "Flows validation doesn't show any discrepancies"
+        [flow1, flow2].each {
+            northbound.validateFlow(it.flowId).each { direction -> assert direction.asExpected }
+        }
+
+        and: "All switches are valid"
+        def involvedSwitches = [fl1SwPair.src, fl1SwPair.dst, fl2SwPair.src, fl2SwPair.dst]*.dpId.unique()
+        Wrappers.wait(RULES_INSTALLATION_TIME) {
+            involvedSwitches.each { swId ->
+                northbound.validateSwitch(swId).verifyRuleSectionsAreEmpty(["missing", "excess", "misconfigured"])
+            }
+        }
+        def switchesAreValid = true
+
+        and: "server42 stats are available for the flow2 in the forward direction"
+        TimeUnit.SECONDS.sleep(STATS_FROM_SERVER42_LOGGING_TIMEOUT + WAIT_OFFSET)
```

Comment:
This should be reworked, we can't afford that long sleep. `STATS_FROM_SERVER42_LOGGING_TIMEOUT` is just a random number that was decided to be used as timeout, it has no real application and you should not rely on it here

Reply:
added `todo` and `low_priority` tag.
Actually don't have any idea now how to rework.